src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,189 @@
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public ProcessResult updateCase(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        boolean ignoreWarnings,
+        String idamToken,
+        String userId,
+        CaseDetails existingCase
+    ) {
+        log.info(
+            "Start updating case for service {} with case ID {} from exception record {}",
+            configItem.getService(),
+            existingCase.getId(),
+            exceptionRecord.id
+        );
+
+        try {
+            String s2sToken = s2sTokenGenerator.generate();
+
+            SuccessfulUpdateResponse updateResponse = caseUpdateClient.updateCase(
+                configItem.getUpdateUrl(),
+                existingCase,
+                exceptionRecord,
+                s2sToken
+            );
+
+            if (!ignoreWarnings && !updateResponse.warnings.isEmpty()) {
+                // do not log warnings
+                return new ProcessResult(updateResponse.warnings, emptyList());
+            }
+
+            log.info(
+                "Successfully called case update endpoint of service {} to update case with case ID {} "
+                    + "based on exception record ref {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            updateCaseInCcd(
+                ignoreWarnings,
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord,
+                updateResponse.caseDetails,
+                existingCase
+            );
+
+            log.info(
+                "Successfully updated case for service {} with case ID {} based on exception record ref {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            return new ProcessResult(
+                exceptionRecordFinalizer.finalizeExceptionRecord(existingCase.getData(), existingCase.getId())
+            );
+        } catch (InvalidCaseDataException exception) {
+            if (BAD_REQUEST.equals(exception.getStatus())) {
+                throw new CallbackException("Failed to update case", exception);
+            } else {
+                return new ProcessResult(exception.getResponse().warnings, exception.getResponse().errors);
+            }
```

Comment:
I realise this pattern is already present for case creation, but there are some problems with it and I think we should make a fix before using CaseUpdateClient.

Here's what we're expecting from update API:
- if the call was successful, return success
- if we sent unprocessable data (well-formed, but business-wise problematic), the update service should return 422 with errors that we should, in turn, return to CCD within a 200 response (so that the caseworkers can read those errors)
- if we sent malformed request, we should get 400 from update service and then return a 500 response to CCD, saying something went wrong, with no specific details

Here we catch `InvalidCaseDataException`, which currently represents a 400/422 response from the update service. This connection is unclear when reading this code. We have to dig deeper to find out what exactly we can expect in the exception. Also, while `BAD_REQUEST` and `UNPROCESSABLE_ENTITY` are the only possibilities at the moment, it's not guaranteed that nobody will at some point reuse it for another 4xx response, introducing a bug.

I think here's what we need to do:
- make `CaseUpdateClient` always throw an exception that strictly represents one response status
- catch those exceptions at the level where we call `CaseUpdateClient`, instead of the whole scope